libvast/src/system/disk_monitor.cpp

```diff
@@ -106,14 +106,14 @@ disk_monitor(disk_monitor_actor::stateful_pointer<disk_monitor_state> self,
   self->state.index = index;
   self->send(self, atom::ping_v);
   return {
-    [self](atom::ping) {
-      self->delayed_send(self, self->state.config.scan_interval, atom::ping_v);
+    [self](atom::ping) -> caf::result<void> {
       if (self->state.purging) {
         VAST_DEBUG("{} ignores ping because a deletion is still in "
                    "progress",
                    *self);
-        return;
+        return caf::skip;
```

Comment:
I believe we should not skip in this case. If we do that means the ping will be handled at the moment we reached the `low_water_mark`, and it doesn't make much sense to check in that moment.